src/main/scala/firrtl/passes/Resolves.scala

```diff
@@ -13,13 +14,18 @@ object ResolveKinds extends Pass with PreservesAll[Transform] {
 
   override val prerequisites = firrtl.stage.Forms.WorkingIR
 
-  type KindMap = collection.mutable.LinkedHashMap[String, Kind]
+  type KindMap = collection.mutable.HashMap[String, Kind]
 
-  def find_port(kinds: KindMap)(p: Port): Port = {
-    kinds(p.name) = PortKind ; p
+  def find_port(kinds: KindMap)(p: Port): Unit = {
```

Comment:
Why not make the new one private?

Reply:
Yeah I didn't do this originally with binary compatibility (forwards nor backwards) in mind. I'll change this to do better practice with making stuff private 🙂